datashader/transfer_functions/__init__.py

```diff
@@ -391,50 +393,71 @@ def _colorize(agg, color_key, how, alpha, span, min_alpha, name, color_baseline,
     data = agg_t.data.transpose([1, 2, 0])
     if da and isinstance(data, da.Array):
         data = data.compute()
-    color_data = data.copy()
+
+    H, W, C = data.shape
+    color_data = ascontigous(data.copy())
+
+    # Keep a NaN mask for later, and a non-NaN mask for averages
+    nan_mask = np.isnan(data)
+    color_mask = ~nan_mask  # bool (H, W, C)
 
     # subtract color_baseline if needed
     with warnings.catch_warnings():
         warnings.filterwarnings('ignore', r'All-NaN slice encountered')
         baseline = np.nanmin(color_data) if color_baseline is None else color_baseline
     with np.errstate(invalid='ignore'):
+        # in-place add/sub to minimize temporaries
         if baseline > 0:
-            color_data -= baseline
+            np.subtract(color_data, baseline, out=color_data, where=color_mask)
```

Comment:
Wasn't this already in place? Is it to avoid calculation along the color_mask axis?